src/scrcmd_20.c

```diff
@@ -0,0 +1,362 @@
+}
+
+static BOOL sub_0204F228(TaskManager *taskManager) {
+    UnkStruct_0204F1E4 *r4 = TaskManager_GetEnv(taskManager);
+    switch (r4->state) {
+        case 0: {
+            if (sub_02037030(0x84, r4, sizeof(UnkStruct_0204F1E4)) == 1) {
+                r4->state++;
+            }
+            break;
+        }
+        case 1: {
+            if (r4->unk01 >= 2) {
+                if (r4->playerSpecies == r4->partnerSpecies) {
+                    *r4->resultPtr = FALSE;
+                } else {
+                    *r4->resultPtr = TRUE;
+                }
+                r4->state++;
+            }
+            break;
+        }
+        case 2: {
+            FreeToHeap(r4);
+            return TRUE;
+        }
+    }
+    return FALSE;
+}
+
+static void sub_0204F284(TaskManager *taskManager, void *a1, u32 challengeType) {
+    FieldSystem *fsys = TaskManager_GetSys(taskManager);
+    UnkStruct_0204F284 *r4 = AllocFromHeap(HEAP_ID_FIELD, sizeof(UnkStruct_0204F284));
+    MI_CpuFill8(r4, 0, sizeof(UnkStruct_0204F284));
+    r4->challengeType = challengeType;
+    r4->unk08 = a1;
+    TaskManager_Call(fsys->taskman, sub_0204F2B8, r4);
+}
+
+static BOOL sub_0204F2B8(TaskManager *taskManager) {
+    FieldSystem *fsys = TaskManager_GetSys(taskManager);
+    UnkStruct_0204F284 *r4 = TaskManager_GetEnv(taskManager);
+    switch (r4->state) {
+        case 0: {
+            r4->state = sub_0204F320(r4, fsys, HEAP_ID_FIELD);
+            break;
+        }
+        case 1: {
+            r4->state = sub_0204F3F8(r4, fsys);
+            break;
+        }
+        case 2: {
+            r4->state = sub_0204F448(r4, fsys, HEAP_ID_FIELD);
+            break;
+        }
+        case 3: {
+            r4->state = sub_0204F4D8(r4, fsys);
+            break;
+        }
+        case 4: {
+            FreeToHeap(r4);
+            return TRUE;
+        }
+    }
+    return FALSE;
+}
+
+static u32 sub_0204F320(UnkStruct_0204F284 *a0, FieldSystem *fsys, HeapID unused) {
+    PartyMenuAppData *partyMenu = AllocFromHeap(HEAP_ID_FIELD, sizeof(PartyMenuAppData));
+    MIi_CpuClearFast(0, (u32*)partyMenu, sizeof(PartyMenuAppData));
+    partyMenu->party = SaveArray_PlayerParty_Get(fsys->savedata);
+    partyMenu->bag = SaveGetBag(fsys->savedata);
+    partyMenu->mailbox = Save_Mailbox_Get(fsys->savedata);
+    partyMenu->options = Save_PlayerData_GetOptionsAddr(fsys->savedata);
+    partyMenu->unk_25 = 0;
+    partyMenu->unk_24 = 22;
+    partyMenu->fsys = fsys;
+    partyMenu->unk_26 = a0->unk05;
+    for (u8 i = 0; i < 2; i++) {
+        partyMenu->unk_30[i] = a0->unk06[i];
+    }
+    partyMenu->unk_37 = 30;
+    partyMenu->unk_36_0 = 1;
+    partyMenu->unk_36_4 = 1;
+    partyMenu->fsys_unk_10C = &fsys->unk_10C;
+    if (a0->challengeType == 1) { // doubles
+        partyMenu->unk_36_0 = 2;
+        partyMenu->unk_36_4 = 2;
+    }
+    FieldSys_LaunchApplication(fsys, &_0210159C, partyMenu);
+    *a0->unk08 = partyMenu;
```

Comment:
please put brackets around deferences like this to make it clear what is being dereferenced

Reply:
also to add on to this you can prolly document unk08 in UnkStruct_0204F284 to be 'PartyMenuAppData *partyMenu' unless something later down the line contradicts that